React/src/Sidebar.js

```diff
@@ -13,13 +13,13 @@ import { _fetch, getFileURL } from './global';
 import './Sidebar.css';
 
 function Sidebar() {
-    const [rooms, setRooms] = useState([]);
-    const [{ user }, dispatch] = useStateValue();
-
+    // const [rooms, setRooms] = useState([]);
+    const [{ user, rooms }, dispatch] = useStateValue();
+    console.log(rooms)
     useEffect(() => {
         _fetch(user, "/api/ChatRoom/GetAll")
             .then(response => response.json())
-            .then(rooms => setRooms(rooms));
+            // .then(rooms => setRooms(rooms));
```

Comment:
As mentioned in previous suggestion, you should dispatch rooms here instead of inside the useEffect() of Chat.js.

```suggestion
            .then((_rooms) => {
                dispatch({
                    type: actionTypes.SET_ROOMS,
                    rooms: _rooms,
                });
            });
```

Reply:
Also regarding this, if I don't get rooms from the store here, then the room will fetch just once and it doesn't update onMessage sent.
Remember in SidebarChat where the room is passed as a prop